sycl/include/sycl/ext/intel/experimental/esimd/detail/test_proxy.hpp

```diff
@@ -0,0 +1,106 @@
+//==-------------- test_proxy.hpp - DPC++ Explicit SIMD API ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Test proxy to differentiate move and copy constructor calls
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#ifndef __ESIMD_ENABLE_TEST_PROXY
+
+// No ABI-breaking changes by default
+#define __ESIMD_DECLARE_TEST_PROXY
+#define __ESIMD_DECLARE_TEST_PROXY_ACCESS
+#define __esimd_move_test_proxy(other)
+
+#else
+
+// Declare the class attribute
+//
+// We are using non static data member initialization approach to force
+// the value required. Initialization will take place even if no
+// default/copy/move constructor of the test_proxy class was explcitly
+// called by any of the user-defined constructors of the proxy target
+#define __ESIMD_DECLARE_TEST_PROXY                                             \
+  esimd::detail::test_proxy M_testProxy = esimd::detail::test_proxy();
+
+// Declare the getter to access the proxy from the tests
+#define __ESIMD_DECLARE_TEST_PROXY_ACCESS                                      \
+  const auto &get_test_proxy() const { return M_testProxy; }
+
+// Test proxy will be handled in a proper way by default/implicit move
+// constructors and move operators.
+// Still the user-defined constructors or move operators should explicitly state
+// what to do with each of class atributes, so a proper wrapper required
+#define __esimd_move_test_proxy(other)                                         \
+  do {                                                                         \
+    M_testProxy = std::move(other.M_testProxy);                                \
+  } while (false);
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace ext {
+namespace intel {
+namespace experimental {
+namespace esimd {
+namespace detail {
+
+/// The test_proxy class.
```

Comment:
Please mention that it is used solely for testing purposes, and there is no extra code generated for usual builds.

Reply:
Thanks, done by 7802416670dcc395c45ab89c2100c95e93d253c1. I've tried to avoid repeating the identical statements though.